src/server/lib/callback-handler.js

```diff
@@ -203,22 +221,60 @@ export default async (sessionToken, profile, providerAccount, options) => {
             providerAccount.refreshToken,
             providerAccount.accessToken,
             providerAccount.accessTokenExpires
-          )
-          await dispatchEvent(events.linkAccount, { user, providerAccount })
+          );
+          await dispatchEvent(events.linkAccount, { user, providerAccount });
 
-          session = useJwtSession ? {} : await createSession(user)
-          isNewUser = true
+          session = useJwtSession ? {} : await createSession(user);
+          isNewUser = true;
           return {
             session,
             user,
-            isNewUser
+            isNewUser,
+          };
+        }
+      }
+    } else if (providerAccount.type === "webhook") {
+      // If signing in with an email, check if an account with the same email address exists already
+      const userByEmail = profile.phoneNumber
+        ? await getUserByAttribute("phoneNumber", profile.phoneNumber)
+        : null;
+
+      if (userByEmail) {
+        // If they are not already signed in as the same user, this flow will
+        // sign them out of the current session and sign them in as the new user
+        if (isSignedIn) {
+          if (user.id !== userByEmail.id && !useJwtSession) {
+            // Delete existing session if they are currently signed in as another user.
+            // This will switch user accounts for the session in cases where the user was
+            // already logged in with a different account.
+            await deleteSession(sessionToken);
           }
         }
+
+        // // Update emailVerified property on the user object
```

Comment:
Need to figure out how to deal with this, maybe have a `verificationProperty` on the `provider` so we know which field to set to true when verification.